# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1272086412 25200
#      Fri Apr 23 22:20:12 2010 -0700
# Node ID ea79ebe99051aadc74cff5d094ab02f405ff0d50
# Parent  0ecbb17d130ad96a595a1dc0cf6996847d716926
Clarify compilation note about defaults.h

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2010-04-23  Rik <octave@nomad.inbox5.com>
+
+	* Makefile.am: Clarify compilation note about defaults.h
+
 2010-04-23  Rik <octave@nomad.inbox5.com>
 
 	* symtab.h: Clear global variables when 'clear -all' called
 
 2010-04-23  John W. Eaton  <jwe@octave.org>
 
 	* DLD-FUNCTIONS/fltk_backend.cc	(OpenGL_fltk::OpenGL_fltk):
 	Explicitly initizlize all data members in the order they are declared.
diff --git a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -732,21 +732,19 @@ endif
 #lex.cc : lex.l
 #	$(LEX) $(LFLAGS) $< > $(@F)-t
 #	@mv $(@F)-t $@
 
 
 $(OPT_INC) : %.h : %.in
 	$(MAKE) -C $(@D) $@
 
-## We want to force an update of defaults.h and oct-conf.h every
-## time make is run because some values may come from the command
-## line or the environment.  The substitution rules use move-if-change,
-## so this should not cause trouble if the file already exists and the
-## newly generated file is not different.
+## defaults.h and oct-conf.h must depend on Makefile.  Calling configure
+## may change default/config values.  However, calling configure will also
+## regenerate the Makefiles from Makefile.am and trigger the rules below.
 
 defaults.h: defaults.h.in Makefile
 	$(do_subst_default_vals)
 
 oct-conf.h: oct-conf.h.in Makefile
 	$(do_subst_config_vals)
 
 oct-errno.cc: oct-errno.cc.in Makefile
